webui/react/src/components/ResourceChart.tsx

```diff
@@ -0,0 +1,94 @@
+import Plotly, { Data, Layout } from 'plotly.js-basic-dist';
+import React from 'react';
+import createPlotlyComponent from 'react-plotly.js/factory';
+
+import { getStateColor, lightTheme } from 'themes';
+import { CommandState, CommonProps, Resource, ResourceState, resourceStates, resourceTypes } from 'types';
+
+const Plot = createPlotlyComponent(Plotly);
+interface Props extends CommonProps {
+  title: string;
+  resources: Resource[];
+}
+
+export interface PlotInfo {
+  data: Data[];
+  layout: Partial<Layout>;
+}
+
+const genPlotInfo = (title: string, resources: Resource[]): PlotInfo | null => {
+
+  const tallies: Record<ResourceState, number> = (() => {
```

Comment:
nit: I think `tally` implies a collection of count, so no need for the plural form